File: 백준/Silver/2531. 회전 초밥/회전 초밥.py
import sys
input = sys.stdin.readline

n,d,k,c=map(int,input().split())
s=[]
for i in range (n):
    s.append(int(input()))

right=k
left=0

answer = 0

while left < n:
  a = set()

  for i in range(left, right):
    a.add(s[i%n])
  if c not in a:
    a.add(c)

  answer = max(answer, len(a))
  left += 1
  right += 1

print(answer)

File: 백준/Silver/2531. 회전 초밥/회전 초밥.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().split('\n');
const [n,d,k,c] =input[0].split(' ').map(Number);
const s = input.slice(1).map(Number);

let right = k;
let left = 0;

let answer = 0;

while (left < n) {
    const a = new Set();
    
    for ( let i = left; i<right; i++){
        a.add(s[i%n]);
    }
    if (! a.has(c)) {
        a.add(c);
    }
    answer = Math.max( answer, a.size);
    left ++;
    right ++;
}

console.log(answer);